src/main/java/uk/gov/register/resources/PreviewRecordsResource.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.register.resources;
+
+import com.codahale.metrics.annotation.Timed;
+import io.dropwizard.views.View;
+import uk.gov.register.core.Record;
+import uk.gov.register.core.RegisterReadOnly;
+import uk.gov.register.views.ViewFactory;
+import uk.gov.register.views.representations.ExtraMediaType;
+
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import java.util.Collections;
+import java.util.List;
+
+@Path("/")
+public class PreviewRecordsResource {
+
+    private static final int DEFAULT_OFFSET = 0;
+
+    protected final ViewFactory viewFactory;
+    private final RegisterReadOnly register;
+
+    @Inject
+    public PreviewRecordsResource(final ViewFactory viewFactory, final RegisterReadOnly register) {
+        this.viewFactory = viewFactory;
+        this.register = register;
+    }
+
+    @GET
+    @Path("/preview-records-json")
+    @Produces(ExtraMediaType.TEXT_HTML)
+    @Timed
+    public View jsonPreview() {
+        return previewPageView(MediaType.APPLICATION_JSON);
+    }
+
+    @GET
+    @Path("/preview-records-yaml")
+    @Produces(ExtraMediaType.TEXT_HTML)
+    @Timed
+    public View yamlPreview() {
+        return previewPageView(ExtraMediaType.TEXT_YAML_TYPE.getSubtype());
+    }
+
+    @GET
+    @Path("/preview-records-ttl")
+    @Produces(ExtraMediaType.TEXT_HTML)
+    @Timed
+    public View ttlPreview() {
+        return previewPageView(ExtraMediaType.TEXT_TTL_TYPE.getSubtype());
+    }
+
+    @GET
+    @Path("/preview-records-json/{record-key}")
+    @Produces(ExtraMediaType.TEXT_HTML)
+    @Timed
+    public View jsonByKeyPreview(@PathParam("record-key") final String key) {
+        return previewByKeyPageView(key, MediaType.APPLICATION_JSON);
+    }
+
+    @GET
+    @Path("/preview-records-yaml/{record-key}")
+    @Produces(ExtraMediaType.TEXT_HTML)
+    @Timed
+    public View yamlByKeyPreview(@PathParam("record-key") final String key) {
+        return previewByKeyPageView(key, ExtraMediaType.TEXT_YAML_TYPE.getSubtype());
+    }
+
+    @GET
+    @Path("/preview-records-ttl/{record-key}")
+    @Produces(ExtraMediaType.TEXT_HTML)
+    @Timed
+    public View ttlByKeyPreview(@PathParam("record-key") final String key) {
+        return previewByKeyPageView(key, ExtraMediaType.TEXT_TTL_TYPE.getSubtype());
+    }
+
+    private View previewPageView(final String mediaType) {
+        final int limit = register.getTotalRecords() > IndexSizePagination.ENTRY_LIMIT
+                ? IndexSizePagination.ENTRY_LIMIT
+                : register.getTotalRecords();
+        final List<Record> records = register.getRecords(limit, DEFAULT_OFFSET);
+
+        return viewFactory.previewRecordsPageView(records, null, mediaType);
+    }
+
+    private View previewByKeyPageView(final String key, final String mediaType) {
+        final List<Record> records = Collections.singletonList(register.getRecord(key).orElseThrow(NotFoundException::new));
```

Comment:
We should be consistent in the errors that we're displaying. In `PreviewItemResource` we display more information to the user, telling them that an item with the key they searched for could not be found, whereas here we give no clue to the user as to what went wrong.